File: src/main/java/fr/gouv/tac/analytics/server/config/AnalyticsProperties.java
package fr.gouv.tac.analytics.server.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.NotEmpty;

@Component
@ConfigurationProperties(prefix = "analytics")
@Validated
@Data
public class AnalyticsProperties {

    @NotEmpty
    private String creationTopic;

    @NotEmpty
    private String deletionTopic;
}

File: src/main/kotlin/gouv/tac/analytics/config/AnalyticsProperties.kt
package fr.gouv.tac.analytics.config

import lombok.Data
import lombok.Value
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.stereotype.Component
import org.springframework.validation.annotation.Validated

@Component
@ConfigurationProperties(prefix = "analytics")
@Validated
@ConstructorBinding
@Value
class AnalyticsProperties(val creationTopic: String = "", val deletionTopic: String = "", val robertJwtAnalyticsPublicKey: String = "")